RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -0,0 +1,202 @@
+//
+//  File.swift
+//  
+//
+//  Created by Josh Holtz on 6/11/24.
+//
+
+import RevenueCat
+import SwiftUI
+// swiftlint:disable all
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private class ComponentPaywallData: ObservableObject {
+    @Published var selectedPackage: TemplateViewConfiguration.Package
+
+    init(selectedPackage: TemplateViewConfiguration.Package) {
+        self._selectedPackage = .init(initialValue: selectedPackage)
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TemplateComponentsView: TemplateViewType {
+
+
+    @StateObject
+    private var componentPaywallData: ComponentPaywallData
+
+    let configuration: TemplateViewConfiguration
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+    @Environment(\.locale)
+    var locale
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
```

Comment:
nitpick: organize stuff as public first, then internal, then private. same for maybe moving `ComponentPaywallData` to the bottom since it's private. 

That way a maintainer will first read through the public stuff, which they'll care about more, then the internal, and they'll only need to read through private stuff if they care about the specifics of the implementation